src/v/cluster/tm_stm.cc

```diff
@@ -532,6 +532,33 @@ ss::future<tm_stm::get_txs_result> tm_stm::get_all_transactions() {
     co_return ans;
 }
 
+ss::future<tm_stm::op_status> tm_stm::delete_partition_from_tx(
+  kafka::transactional_id tid, tm_transaction::tx_partition ntp) {
+    if (!_c->is_leader()) {
+        co_return tm_stm::op_status::not_leader;
+    }
+
+    auto is_ready = co_await sync(_sync_timeout);
+    if (!is_ready) {
+        co_return tm_stm::op_status::unknown;
+    }
+
+    auto tx_it = _mem_txes.find(tid);
+    if (tx_it == _mem_txes.end()) {
+        tx_it = _log_txes.find(tid);
+        if (tx_it == _log_txes.end()) {
+            co_return tm_stm::op_status::not_found;
+        }
+    }
+
+    auto res = tx_it->second.delete_partition(ntp);
```

Comment:
Should I also delete partition from log state if transaction is both containers (mem state and log state)

Reply:
Good question. I think @rystsov will need to answer this. My best guess is that it should be removed from both (unless existence in one container like mem_txes implies it isn't in the other one already).